src/sentry/api/endpoints/organization_dashboards.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+
+from django.db import IntegrityError, transaction
+from sentry.api.base import DocSection
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.bases.dashboard import DashboardSerializer
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.serializers import serialize
+from sentry.models import Dashboard
+from rest_framework.response import Response
+
+
+class OrganizationDashboardsEndpoint(OrganizationEndpoint):
+    doc_section = DocSection.ORGANIZATIONS
+
+    def get(self, request, organization):
+        """
+        Retrieve an Organizations Dashboards
+        `````````````````````````````````````
+        Retrieve a list of dashboards that are associated with the given organization.
+        :pparam string organization_slug: the slug of the organization the
+                                          dashboards belongs to.
+        :qparam string query: the title of the dashboard being searched for.
+        :auth: required
+        """
+        dashboards = Dashboard.objects.filter(
+            organization_id=organization.id
+        )
+        query = request.GET.get('query')
+        if query:
+            dashboards = dashboards.filter(
+                title__icontains=query,
```

Comment:
Does this column have an index on it? Does it need one?

Reply:
@markstory it currently doesn't. I kind of assume that the number of dashboards in a given organization won't be extremely large. I could be wrong on that. What do you think? Also cc @wedamija

Reply:
Actually @markstory I was wrong. We have this unquie together constrain that search should hit. https://github.com/getsentry/sentry/blob/master/src/sentry/models/dashboard.py#L28

Reply:
Yeah that index will cover it. I imagine most orgs won't have too many dashboards, but it's not an issue since we have that unique index in place anyway!

Reply:
Oh, just realized. Since we're using `icontains` the title portion of the index will likely not be used, since it does a query like `organization_id = 1 and title LIKE '%mystring%'`. This still shouldn't be an issue IMO, since I think the number of dashboards will be reasonably low.